src/core/painting/qgsgeometrypaintdevice.cpp

```diff
@@ -320,15 +336,31 @@ void QgsGeometryPaintEngine::drawPolygon( const QPointF *points, int pointCount,
   else
   {
     const QTransform transform = painter()->combinedTransform();
-    drawPolygonImp( transform, mGeometry, points, pointCount, mode );
+    drawPolygonImp( transform, mGeometry, points, pointCount, mode, mSimplifyTolerance );
   }
 }
 
 void QgsGeometryPaintEngine::addStrokedLine( const QgsLineString *line, double penWidth, Qgis::EndCapStyle endCapStyle, Qgis::JoinStyle joinStyle, double miterLimit, const QTransform *matrix )
 {
-  QgsGeos geos( line );
+  std::unique_ptr< QgsAbstractGeometry > buffered;
+  if ( mSimplifyTolerance > 0 )
+  {
+    // For performance, we apply a lower level of simplification to the line BEFORE doing the buffer.
```

Comment:
I'm afraid that it doesn't improve [this issue](https://github.com/qgis/QGIS/pull/57623#issuecomment-2147639769)

Reply:
Can you retest please? There's no way the gridded/pixelated mask output could be exported by this branch.

Reply:
Way better

painterpath way
![simplifybefore](https://github.com/qgis/QGIS/assets/14358135/8817061d-913f-4b67-96bd-38276ba89551)

geometry way
![simplify](https://github.com/qgis/QGIS/assets/14358135/8e165e31-8267-4d90-b9a7-9ced3e3694d3)

Can we make it a little less agressive by default to have a prettier renderering

Reply:
@troopa81 

Thanks for the retest!

> Can we make it a little less agressive by default to have a prettier renderering

Let's revisit that after https://github.com/qgis/QGIS/pull/57799 -- with this PR alone the export file sizes are still very large